internal/filesystem.go

```diff
@@ -107,6 +108,43 @@ func (rdl *runDirListener) Addr() net.Addr {
 // ManualListener returns manually created listener with provided permissions
 func ManualListener(socket string, perm fs.FileMode) func() (net.Listener, error) {
 	return func() (net.Listener, error) {
+		pidfile := DaemonPid
```

Comment:
Please move pid logic to a separate function. and use it here

Reply:
Ok, done